code/game/objects/structures/safe.dm

```diff
@@ -6,50 +6,70 @@ SAFE INTERNALS
 SAFE CODES
 */
 
+#define DRILL_SPARK_CHANCE 15
+#define DRILL_TIME 300 SECONDS
+#define SOUND_CHANCE 10
+
 GLOBAL_LIST_EMPTY(safes)
 
-//SAFES
+/**
+  * # Safe
+  *
+  * A locked container that can only be opened by entering a combination through a dial.
+  */
 /obj/structure/safe
-	name = "\improper Safe"
+	name = "safe"
 	desc = "A huge chunk of metal with a dial embedded in it. Fine print on the dial reads \"Scarborough Arms tumbler safe, guaranteed thermite resistant, explosion resistant, and assistant resistant.\""
 	icon = 'icons/obj/structures.dmi'
 	icon_state = "safe"
-
 	anchored = TRUE
 	density = TRUE
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
-
+	// Settings
+	/// The maximum combined w_class of stuff in the safe.
+	var/maxspace = 24
+	/// The amount of tumblers that will be generated.
+	var/number_of_tumblers = 3
+	// Variables
+	/// Whether the safe is open or not.
 	var/open = FALSE
+	/// Whether the safe is locked or not.
 	var/locked = TRUE
-	var/dial = 0		// The position the dial is pointing to.
-
-	var/number_of_tumblers = 3 // The amount of tumblers that will be generated.
-	var/list/tumblers = list() // The list of tumbler dial positions that need to be hit.
-	var/current_tumbler_index = 1 // The index in the tumblers list of the tumbler dial position that needs to be hit.
-
-	var/space = 0		// The combined w_class of everything in the safe.
-	var/maxspace = 24	// The maximum combined w_class of stuff in the safe.
-
+	/// The position the dial is pointing to.
+	var/dial = 0
+	/// The list of tumbler dial positions that need to be hit.
+	var/list/tumblers = list()
+	/// The index in the tumblers list of the tumbler dial position that needs to be hit.
+	var/current_tumbler_index = 1
+	/// The combined w_class of everything in the safe.
+	var/space = 0
+	// Variables (Drill)
+	/// The currently placed thermal drill, if any.
 	var/obj/item/thermal_drill/drill = null
+	/// The [/proc/addtimer] handle for the current thermal drill.
 	var/drill_timer
```

Comment:
If this is a timer object shouldnt it be typecasted to one?

Reply:
`addtimer()` returns the timer handle's *ID* rather than the handle itself.

https://github.com/ParadiseSS13/Paradise/blob/56d762bafbf4685bbdbb46a7d772042be9c2d5fb/code/controllers/subsystem/timer.dm#L501-L502